Updated mine() with proof of work, and associated tests

jldistefano opened the pull request.

kstola2 commented:
Why did you delete `block.cpython-36.pyc` ? It might be causing some of the merge conflicts.

kstola2 commented:
I'm still getting a merge conflict. Can you try doing a rebase-pull to update your repo? That might fix it.

jldistefano commented:
I've readded block.cpython-36.pyc to my fork.

kstola2 commented:
Do you get any problems if you run `./merge_with_main_repository.sh` in your bash terminal while in the folder?

jldistefano commented:
Yes, it says Permissions Denied

On Thu, Nov 1, 2018 at 11:35 PM kstola2 <notifications@github.com> wrote:

> Do you get any problems if you run ./merge_with_main_repository.sh in
> your bash terminal while in the folder?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SIGBlockchain/Project/pull/64#issuecomment-435268707>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ApcKcsF-ceIuairrnd6sX-Czj60L9rSuks5uq8uMgaJpZM4YJR0p>
> .
>

kstola2 commented:
There shouldn't be any merge conflicts, which is weird. One thing I've done before is copy my additions to the file, delete my repository off my github, refork it from the main repo, then paste in the code i copied and try again. The problem is that we've had this issue happen before and it sort of screwed everybody elses' pull requests.

jldistefano commented:
I just backed up my local block.py and block.test.py files, deleted the
rest of the project. Cloned the project directly from the main repo, copied
the changes manually, and updated my fork.
There was still a merge conflict, so I just looked at the conflict and
deleted the problem section. Every function is there and it should be ok to
merge now.

On Fri, Nov 2, 2018 at 12:02 AM kstola2 <notifications@github.com> wrote:

> There shouldn't be any merge conflicts, which is weird. One thing I've
> done before is copy my additions to the file, delete my repository off my
> github, refork it from the main repo, then paste in the code i copied and
> try again. The problem is that we've had this issue happen before and it
> sort of screwed everybody elses' pull requests.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SIGBlockchain/Project/pull/64#issuecomment-435271878>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ApcKchFNBn6Doo_I58ECcTeN7It3hysMks5uq9HKgaJpZM4YJR0p>
> .
>

jldistefano commented:
Requested changes added

On Fri, Nov 2, 2018 at 12:29 AM kstola2 <notifications@github.com> wrote:

> *@kstola2* requested changes on this pull request.
>
> I have a few questions I made some single comments.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SIGBlockchain/Project/pull/64#pullrequestreview-170974135>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ApcKcrpnsz5bxVJk1kaD-du0cg9HHUR4ks5uq9gqgaJpZM4YJR0p>
> .
>

kstola2 commented:
So I think I may made a gross miscalculation when we decided earlier to forego the hexlify in the hashSHA function, which is totally on me. This is causing a lot of issues that I'll need to look into tomorrow and fix. Your code is fine, but because of this problem I'll need to change a few things first before approving the changes. It's late though so I'll get to it tomorrow.
